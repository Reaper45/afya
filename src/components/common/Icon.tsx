import * as React from 'react'
import { ImgHTMLAttributes } from 'react'

import styled from 'lib/emotion'

const IconWrapper = styled("img")<{ active?: boolean }>`
  color: ${props =>
    props.active ? props.theme.colors.accent : props.theme.colors.primary};
`;

const Icon: React.FC<ImgHTMLAttributes<HTMLImageElement> & {
  active?: boolean;
}> = props => {
  return <IconWrapper {...props} />;
};

export default Icon;
