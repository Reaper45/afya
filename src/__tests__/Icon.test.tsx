import React from "react";
import { shallow } from "enzyme";

import Icon from "components/common/Icon";

const src = "static/file.svg"

it("render icon correctly", () => {
  const component = shallow(<Icon src={src} />);
  expect(component).toMatchSnapshot();
});

it("render active icon", () => {
  const component = shallow(<Icon src={src} active />);
  expect(component).toMatchSnapshot();
});