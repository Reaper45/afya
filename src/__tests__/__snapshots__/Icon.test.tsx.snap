// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render active icon 1`] = `
<Styled(img)
  active={true}
  src="static/file.svg"
/>
`;

exports[`render icon correctly 1`] = `
<Styled(img)
  src="static/file.svg"
/>
`;
